x-pack/plugins/fleet/server/services/agents/reassign.ts

```diff
@@ -80,10 +80,15 @@ export async function reassignAgents(
   options: ({ agents: Agent[] } | GetAgentsOptions) & { force?: boolean },
   newAgentPolicyId: string
 ): Promise<{ items: BulkActionResult[] }> {
-  const agentPolicy = await agentPolicyService.get(soClient, newAgentPolicyId);
-  if (!agentPolicy) {
+  const newAgentPolicy = await agentPolicyService.get(soClient, newAgentPolicyId);
+  if (!newAgentPolicy) {
     throw Boom.notFound(`Agent policy not found: ${newAgentPolicyId}`);
   }
+  if (newAgentPolicy.is_managed) {
```

Comment:
moved the `newAgentPolicy` check at the top, this was also queried in a loop for each agent